tsdb/wal/watcher.go

```diff
@@ -214,8 +244,9 @@ func (w *Watcher) loop() {
 }
 
 // Run the watcher, which will tail the WAL until the quit channel is closed
-// or an error case is hit.
-func (w *Watcher) Run() error {
+// or an error case is hit. Use -1 for startSegment if the watcher should not
+// start tailing samples until after startTime.
+func (w *Watcher) Run(startSegment int) error {
```

Comment:
Don't segment files start at 1? You might be able to use 0 instead of -1 here.

Reply:
Nope, they start at 0
```
ls data/wal
00000000
```

Reply:
👍 my bad :)